docs/specs/xref.yml

```diff
@@ -45,6 +45,56 @@ outputs:
   docs/a.json: |
     { "conceptual": "<p>Link to <a class=\"no-loc\" href=\"/dotnet/api/system.string\">System.String</a></p>\n"}
 ---
+# Remove external xref url hostname 
+# when config hostname is old one but xref url hostname is matched new one
+inputs:
+  docfx.yml: |
+    hostName: docs.microsoft.com
+    hostNameMapping: 
+      docs.microsoft.com: new-docs.microsoft.com
+    extend: _shared/extend.json
+  _shared/extend.json: |
+    {"xref":["1.xrefmap.json"]}
+  docs/a.md: Link to @System.String?displayProperty=fullName
+  1.xrefmap.json: |
+    {
+      "references":[{
+          "uid": "System.String",
+          "name": "String",
+          "fullName": "System.String",
+          "href": "https://new-docs.microsoft.com/dotnet/api/system.string",
+          "nameWithType": "System.String"
+      }]
+    }
+outputs:
+  docs/a.json: |
+    { "conceptual": "<p>Link to <a class=\"no-loc\" href=\"/dotnet/api/system.string\">System.String</a></p>\n"}
+---
+# Remove external xref url hostname 
+# when config hostname is new one but xref url hostname is matched old one
+inputs:
+  docfx.yml: |
+    hostName: new-docs.microsoft.com
+    hostNameMapping: 
+      docs.microsoft.com: new-docs.microsoft.com
+    extend: _shared/extend.json
+  _shared/extend.json: |
+    {"xref":["1.xrefmap.json"]}
+  docs/a.md: Link to @System.String?displayProperty=fullName
+  1.xrefmap.json: |
+    {
+      "references":[{
+          "uid": "System.String",
+          "name": "String",
+          "fullName": "System.String",
+          "href": "https://docs.microsoft.com/dotnet/api/system.string",
```

Comment:
There is no mapping matched in `hostNameMapping`. We should still output `new-docs.microsoft.com`

Reply:
same as above